woocommerce/google-listings-and-ads
# Implement the deletion feature of paid campaigns for the dashboard page

### Changes proposed in this Pull Request:

Closes #595

- Implement the deletion feature of paid campaigns for the dashboard page

### Screenshots:

#### 🎥 . Successfully delete a paid campaign
![Kapture 2021-05-17 at 16 37 12](https://user-images.githubusercontent.com/17420811/118459460-5a855c00-b72e-11eb-8824-30dbedd10557.gif)

#### 📷 . The modal would disable controllable elements and unable to dismiss during deletion calling

![image](https://user-images.githubusercontent.com/17420811/118460843-a97fc100-b72f-11eb-99ea-e2484de0bfa9.png)

#### 📷 . If an error occurred, show an error message at the bottom and keep the confirmation modal open
![image](https://user-images.githubusercontent.com/17420811/118459435-56f1d500-b72e-11eb-9aa5-3585dc6be059.png)

### Detailed test instructions:

1. Head to the dashboard page: `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard`
2. Click on the **Remove** action any program on the program table
3. Click on the **Remove Campaign** button to confirm the deletion
4. The paid campaign should be removed from the program table after API call finished

### Additional note:

~~❗  The API of **DELETE** `/wc/gla/ads/campaigns/<program_id>` responds 400 error with the message "Error deleting campaign: Request contains an invalid argument." And looks like something went wrong here:~~ Fixed by @mikkamp 

https://github.com/woocommerce/google-listings-and-ads/blob/08ee62f4d2a5ef3450f3150760b9f24dfa6478cc/src/API/Google/AdsGroup.php#L260-L263
